### ISSUE TITLE
Issue appearance with sql-db module version 8.0.0

### ISSUE THREAD

**Issue by rogeriaportilho** [terraform-google-modules/terraform-google-sql-db#262]
### TL;DR

Update the terraform version from 0.14.3 to 1.0.7

### Expected behavior

I was expected the updated of terraform version was worked fine. When I update the version from 0.14.3 to 1.0.7 the terraform force to update the sql-db module from 7.2.0 to 8.0.0, but after that I'm deal with some issue.
The version of service provide is 3.89 and beta 3.90.1

### Observed behavior

Error: Error when reading or editing SQL User "admin" in instance "client": googleapi: Error 403: The client is not authorized to make this request., notAuthorized

  with module.cloudsql_instances.module.mysql[1].module.safer_mysql.google_sql_user.default[0],
  on .terraform/modules/cloudsql_instances.mysql/modules/mysql/main.tf line 173, in resource "google_sql_user" "default":
 173: resource "google_sql_user" "default" ***

Error: Error when reading or editing SQL User "admin" in instance "encap-service": googleapi: Error 403: The client is not authorized to make this request., notAuthorized

  with module.cloudsql_instances.module.mysql[0].module.safer_mysql.google_sql_user.default[0],
  on .terraform/modules/cloudsql_instances.mysql/modules/mysql/main.tf line 173, in resource "google_sql_user" "default":
 173: resource "google_sql_user" "default" ***

Error: Error when reading or editing SQL User "admin" in instance "consumer": googleapi: Error 403: The client is not authorized to make this request., notAuthorized

  with module.cloudsql_instances.module.mysql[2].module.safer_mysql.google_sql_user.default[0],
  on .terraform/modules/cloudsql_instances.mysql/modules/mysql/main.tf line 173, in resource "google_sql_user" "default":
 173: resource "google_sql_user" "default" ***

### Terraform Configuration

```hcl
from module
module "private_service_access" {
  source      = "GoogleCloudPlatform/sql-db/google//modules/private_service_access"
  version     = "8.0.0"
  project_id  = var.project_id
  vpc_network = var.vpc_network
}

module "mysql" {
  source              = "GoogleCloudPlatform/sql-db/google//modules/safer_mysql"
  version             = "8.0.0"
  count               = length(var.db_instances)
  project_id          = var.project_id
  region              = var.region
  zone                = var.zones[0]
  deletion_protection = false
  database_flags = [{
    name  = "general_log",
    value = "on"
  }]
  assign_public_ip = var.assign_public_ip

  name             = var.db_instances[count.index].name
  db_name          = var.db_instances[count.index].db_name
  database_version = "MYSQL_12_0"
  tier             = "db-n1-standard-1"

  user_name     = var.db_instances[count.index].root_user
  user_password = var.db_instances[count.index].root_password

  vpc_network       = var.vpc_network_link
  module_depends_on = [module.private_service_access.peering_completed]
}

From main.tf of resource
module "cloudsql_instances" {
  source           = "../../modules/cloudsql"
  project_id       = var.project_id
  region           = var.region
  zones            = var.zones
  vpc_network      = module.vpc_network.network_name
  vpc_network_link = module.vpc_network.network_link
  db_instances     = var.db_instances
}


module "cloud_proxy_sql_sa" {
  source        = "../../modules/gcp_sa"
  project_id    = var.project_id
  names         = var.cloudsql_proxy_config["cloudsql_proxy_sa_names"]
  project_roles = var.cloudsql_proxy_config["cloudsql_proxy_sa_roles"]
  display_name  = var.cloudsql_proxy_config["cloudsql_proxy_sa_display_name"]
}

module "cloud_proxy_sql_sa_iam_binding" {
  for_each         = var.sa_iam_binding
  source           = "../../modules/gcp_iam_binding"
  service_accounts = each.value["service_accounts"]
  project_id       = var.project_id
  iam_bindings     = each.value["iam_bindings"]
}
```

### Terraform Version

```sh
Terraform 1.0.7
```

### Additional information

I considered the tag of bug, but I'm not sure if it is some bug of module

**Comment by bharathkkb** [comment]
Hi @rogeriaportilho 

> the terraform force to update the sql-db module from 7.2.0 to 8.0.0

I dont see a tag for `7.2.0` in this repo and I am not sure why that is the case if its `7.1.0`. `safer_mysql` module [allows](https://github.com/terraform-google-modules/terraform-google-sql-db/blob/f6bdeee3830db33a2e3f0036e75e0931129bf1b0/modules/safer_mysql/versions.tf#L18) any TF version >0.13.

> The client is not authorized to make this request., notAuthorized

This seems like something is wrong when the provider tried to authenticate. How are you configuring the provider to authenticate to GCP?

**Comment by rogeriaportilho** [comment]
Hi @bharathkkb, 
Thank you for your message.
This is the tag https://github.com/terraform-google-modules/terraform-google-sql-db/commit/0365f1794e148bf5f0df487b68fa4640dc17987f about version v7.2.0. However, when you check in the release page I can't see the release 7.2.0. There are 7.1.0 and 8.0.0. I tried to do with the version 7.1.0 but the terraform version force I change the version to 8.0.0. No clue it, because with the version 7.2.0 it was worked.

On base o the permissions it's fine the terraform service account is owner cloud and it already added in the API Cloud SQL Admin.

What does it means "How are you configuring the provider to authenticate to GCP?" ? I'm not sure if you talk about the service account.

**Comment by morgante** [comment]
version 7.2.0 was never released. Instead, we released 8.0.0 directly. You can see this: https://github.com/terraform-google-modules/terraform-google-sql-db/pull/247

We have no intention of support v7.2.0. Please switch to using 8.0.0.

**Comment by rogeriaportilho** [comment]
Hi @morgante, 
First of all, thank you for your message.
In fact, We already used version 7.2.0 without issue, but When we updated the version of from 0.14.3 to 1.0.7 that force to update this module to 8.0.0. However, with this version I'm seeing the message errors:

If you have any tips to fix please, let me know.

Error: Error when reading or editing SQL User "admin" in instance "client": googleapi: Error 403: The client is not authorized to make this request., notAuthorized

with module.cloudsql_instances.module.mysql[1].module.safer_mysql.google_sql_user.default[0],
on .terraform/modules/cloudsql_instances.mysql/modules/mysql/main.tf line 173, in resource "google_sql_user" "default":
173: resource "google_sql_user" "default" ***

Error: Error when reading or editing SQL User "admin" in instance "encap-service": googleapi: Error 403: The client is not authorized to make this request., notAuthorized

with module.cloudsql_instances.module.mysql[0].module.safer_mysql.google_sql_user.default[0],
on .terraform/modules/cloudsql_instances.mysql/modules/mysql/main.tf line 173, in resource "google_sql_user" "default":
173: resource "google_sql_user" "default" ***

Error: Error when reading or editing SQL User "admin" in instance "consumer": googleapi: Error 403: The client is not authorized to make this request., notAuthorized

with module.cloudsql_instances.module.mysql[2].module.safer_mysql.google_sql_user.default[0],
on .terraform/modules/cloudsql_instances.mysql/modules/mysql/main.tf line 173, in resource "google_sql_user" "default":
173: resource "google_sql_user" "default" ***

**Comment by morgante** [comment]
It seems like you're not authenticated correctly. Please make sure you have proper service account permissions.

**Comment by rogeriaportilho** [comment]
@morgante, I already check many times if the process of authentication was correct. Thank you for message. I will work in another way.
